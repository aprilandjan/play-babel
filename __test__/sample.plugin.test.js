const babel = require('@babel/core');
const plugin = require('../sample.plugin');

//  no return;
const example1 = `
function abc() {
  for (var i = 0; i < 100; i++) {
    console.log(i);
  }
}
`;

//  has a return at the end
const example2 = `
function def() {
  for (var i = 0; i < 100; i++) {
    console.log(i);
  }
  return 'done';
}
`;

//  has multiple possible return
const example3 = `
function joker() {
  for (var i = 0; i < 100; i++) {
    console.log(i);
    if (Math.random() < 0.5) {
      return 'abort';
    }
  }
  return 'done';
}
`

//  not direct return for it
const example4 = `
function terrible() {
  const log = () => {
    return 'log!';
  }
  for (var i = 0; i < 100; i++) {
    console.log(i);
    if (Math.random() < 0.5) {
      return 'abort';
    }
  }
  return 'done';
}
`

const transform = (source) => babel.transform(source, {
  plugins: [plugin]
}).code;


describe('sample.plugin.js', () => {
  it('should transform example1 correctly', () => {
    const result = transform(example1);
    expect(result).toMatchSnapshot();
  });
  it('should transform example2 correctly', () => {
    const result = transform(example2);
    expect(result).toMatchSnapshot();
  });
  it('should transform example3 correctly', () => {
    const result = transform(example3);
    expect(result).toMatchSnapshot();
  });
  it.only('should transform example4 correctly', () => {
    const result = transform(example4);
    expect(result).toMatchSnapshot();
  });
});
