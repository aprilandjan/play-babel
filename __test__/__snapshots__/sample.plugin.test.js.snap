// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample.plugin.js should transform example1 correctly 1`] = `
"function abc() {
  console.time(\\"abc\\");

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  console.timeEnd(\\"abc\\")
}"
`;

exports[`sample.plugin.js should transform example2 correctly 1`] = `
"function def() {
  console.time(\\"def\\");

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  console.timeEnd(\\"def\\")
  return 'done';
}"
`;

exports[`sample.plugin.js should transform example3 correctly 1`] = `
"function joker() {
  console.time(\\"joker\\");

  for (var i = 0; i < 100; i++) {
    console.log(i);

    if (Math.random() < 0.5) {
      console.timeEnd(\\"joker\\")
      return 'abort';
    }
  }

  console.timeEnd(\\"joker\\")
  return 'done';
}"
`;

exports[`sample.plugin.js should transform example4 correctly 1`] = `
"function terrible() {
  console.time(\\"terrible\\");

  function log() {
    console.time(\\"log\\");
    console.timeEnd(\\"log\\")
    return 'log';
  }

  for (var i = 0; i < 100; i++) {
    console.log(i);

    if (Math.random() < 0.5) {
      console.timeEnd(\\"terrible\\")
      return 'abort';
    }
  }

  console.timeEnd(\\"terrible\\")
  return 'done';
}"
`;
